techniques/android/MASTG-TECH-0109.md

```diff
@@ -0,0 +1,51 @@
+--- 
+title: Intercepting Flutter HTTP Traffic
+platform: android 
+---
+
+Flutter is an open-source UI software development kit (SDK) created by Google. It is used for building natively compiled applications for mobile, web, and desktop from a single codebase. Flutter uses Dart, which is not proxy-aware and uses its own certificate store. The application doesn't take proxy configuration from the system and send the data directly to the server. Due to this, it is not possible to intercept the request using the BurpSuite or any MITM tools.
+
+
+
+**How does re-flutter method differs from other techniques ?**
+
+There are alternative methods for intercepting traffic, such as [sending traffic to the proxy through ProxyDroid/iptables](https://blog.nviso.eu/2019/08/13/intercepting-traffic-from-android-flutter-applications/). However, these techniques require some configuration. By employing the re-flutter command-line tool, the application can be patched effortlessly without the need for any setup.
+
+## Intercepting Traffic using re-flutter
+
+1. Patch the app to enable traffic interception.
+
+Run the command to patch the app and select the option **Traffic monitoring and interception** and then the IP of the machine which the interception proxy is running.
+```
+$ reflutter demo.apk
+
+Choose an option:
+
+    Traffic monitoring and interception
+    Display absolute code offset for functions
+
+[1/2]? 1
+
+Example: (192.168.1.154) etc.
+Please enter your BurpSuite IP: 192.168.29.216
+```
+
+This will create a **release.RE.apk** file in the output folder.
+
+2. Sign the patched **release.RE.apk** file (e.g. using the [uber-apk-signer](https://github.com/patrickfav/uber-apk-signer)).
+
+```
+   $ java -jar uber-apk-signer.jar -a release.RE.apk --out demo-signed
+```
+
+This will create a **release.RE-aligned-debugSigned.apk** file in the output folder.
+
+3. Install the signed patched app on the mobile device.
+
+4. Configure the interception proxy.For example, in Burp-suite:
```

Comment:
```suggestion
4. Configure the interception proxy. For example, in Burp:
```
We use 'Burp' as the consistent name for PortSwigger's Burp Suite throughout the MASTG.